Add possibility to exclude folders from folder level cyclic deps

gtktsc opened the pull request:
This PR introduces possibility to exclude specified in stricter/circular-dependencies config registries folders from folder-level checks. It allows to create for example registry of reusable components packages or utils. 
One option in stricter/circular-dependencies is added: registries: string[]. It's not a regex, because each time we want to exclude single folder only.
I've also: added tests to make sure I won't break anything, updated readme and fixed link in message.

Li0liQ commented on src/default-rules/circular-dependencies/folder-dependency-graph.ts:
```diff
@@ -71,6 +71,10 @@ export default (fileDependencyGraph: Graph) => {
 
         parentFolders.pop(); // ditch common root
         parentFolders.forEach(parent => {
+            if (registries.some(entry => parent.includes(entry))) {
```
Which parents is it going to exclude?
I have three concerns here:
1. We should still be checking for circular references within the registries
2. The parent matching should be os-independent
3. According to my understanding, `parent.includes` is string matching and it could potentially result in incorrect matches e.g. `'foobar'.includes('foo')`

gtktsc replied:
1. Fixed,
2. Using micromatch now,
3. Micromatch should take care of it

gtktsc commented:
Update: added micromatch to handle paths, right now nested cyclic dependencies will be reported. Fix is covered with new tests. I've also added types for micromatch, but some of them are invalid - added comment to explain.

Li0liQ commented:
Could you, please, update https://github.com/stricter/stricter/blob/master/CHANGELOG.md to mention the change? Please, mention, it's a breaking one due to not popping top-level root folder any more in the rule.

gtktsc commented:
> Could you, please, update https://github.com/stricter/stricter/blob/master/CHANGELOG.md to mention the change? Please, mention, it's a breaking one due to not popping top-level root folder any more in the rule (people will need to specify it explicitly to preserve original behaviour).

It's not breaking old popping behaviour, this should be kept:
https://github.com/stricter/stricter/pull/63/files#diff-caf9abd8dcd2fa2f869c0c93d9e0ae47R49

I'll update https://github.com/stricter/stricter/blob/master/CHANGELOG.md, thanks!

Li0liQ commented:
Let's break it now.
For a greenfield project using the rule this will hep avoid circular references from day one.
Otherwise, people might unknowingly introduce them and fixing cycles post-factum is quite painful.
